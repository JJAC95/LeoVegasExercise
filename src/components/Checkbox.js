import '../styles/checkbox.css';
export const Checkbox = ({ value, label, checked }) => {
	return (
		<div key={value} className="container">
			<input
				id={label}
				defaultValue={value}
				type="checkbox"
				defaultChecked={checked}
				className="checkBox"
			/>
			<label>{label}</label>
		</div>
	);
};
