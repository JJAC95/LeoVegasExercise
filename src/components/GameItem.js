import DefaultImage from '../images/Slot-Machine.jpg';
import '../styles/gameItem.css';
export const GameItem = ({ game }) => {
	const handleError = (e) => {
		e.target.src = DefaultImage;
	};

	return (
		<li>
			<div className="imgContainer">
				<img
					alt={game.gameName}
					src={game.gameThumbnail || DefaultImage}
					onError={handleError}
				/>
			</div>
			<p>{game.gameName}</p>
		</li>
	);
};
