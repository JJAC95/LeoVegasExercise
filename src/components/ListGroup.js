import '../styles/listGroup.css';
export const Listgroup = ({ children }) => {
	return <ul>{children}</ul>;
};
