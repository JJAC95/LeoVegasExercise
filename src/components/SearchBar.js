import '../styles/searchbar.css';

export const SearchBar = ({ style, handleInput }) => {
	return (
		<div className="searchBarContainer">
			<div style={{ position: 'relative' }}>
				<input
					type="text"
					name="searchBar"
					id="input"
					placeholder="Search Games"
					style={style}
					onChange={(input) => handleInput(input)}
				/>
			</div>
		</div>
	);
};
