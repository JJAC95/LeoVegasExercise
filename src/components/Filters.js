import { Checkbox } from './Checkbox';
import '../styles/filters.css';
export const Filters = ({ games, setFilters }) => {
	const handleCheckbox = (e) => {
		let index = games.findIndex((item) => item.name === e.target.value);
		games[index].checked = !games[index].checked;
		setFilters(games.filter((cats) => cats.checked === true));
	};

	return (
		<>
			{games.length === 0 ? (
				<p> Loading </p>
			) : (
				<fieldset onChange={handleCheckbox}>
					{games.map((category) => (
						<Checkbox
							value={category.name}
							label={
								category.name.charAt(0).toUpperCase() + category.name.slice(1)
							}
							checked={category.checked}
							key={category.name}
						/>
					))}
				</fieldset>
			)}
		</>
	);
};
