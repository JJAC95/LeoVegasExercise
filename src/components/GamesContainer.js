import { GameItem } from './GameItem';
import { Listgroup } from './ListGroup';
import '../styles/gameContainer.css';
export const GamesContainer = ({ games }) => {
	return (
		<div className="gamesListContainer">
			<p>All Games</p>
			{games.length === 0 && <p>Sorry no search results found.</p>}
			<Listgroup>
				{games.map((game) => (
					<GameItem game={game} key={game.slug} />
				))}
			</Listgroup>
		</div>
	);
};
