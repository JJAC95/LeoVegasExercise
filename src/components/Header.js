import { SearchBar } from './SearchBar';
import '../styles/header.css';
export const Header = ({ handleSearchInput }) => {
	return (
		<>
			<div className="headerItems">
				<div className="searchSection">
					<SearchBar handleInput={handleSearchInput} />
				</div>
			</div>
		</>
	);
};
