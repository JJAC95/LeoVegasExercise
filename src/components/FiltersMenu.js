import { Disclosure, Transition } from '@headlessui/react';
import { Filters } from './Filters';
import '../styles/filtersMenu.css';
export const FiltersMenu = ({ games, setFilters }) => {
	return (
		<div className="discContainer">
			<Disclosure
				as="section"
				arial-labelledby="filter-heading"
				className="disclosure"
			>
				<div className="discHeaderContainer">
					<div className="discButtonContainer">
						<Disclosure.Button className="discButton">
							Filters
						</Disclosure.Button>
					</div>
				</div>
				<Transition
					enter="transitionOpen"
					enterFrom="transitionOpenFrom"
					enterTo="transitionOpenTo"
					leave="transitionClose"
					leaveFrom="transitionCloseFrom"
					leaveTo="transitionCloseTo"
				>
					<Disclosure.Panel className="panel">
						<Filters games={games} setFilters={setFilters} />
					</Disclosure.Panel>
				</Transition>
			</Disclosure>
		</div>
	);
};
