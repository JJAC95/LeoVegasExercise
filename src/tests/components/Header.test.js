import { render, screen } from '@testing-library/react';
import { Header } from '../../components/Header';

describe('Header Component', () => {
	it('renders correctly', () => {
		const { container } = render(<Header />);

		expect(container).toMatchSnapshot();
	});

	it('shows the searchbar', () => {
		render(<Header />);

		const input = screen.getByPlaceholderText('Search Games');

		expect(input).toBeInTheDocument();
	});
});
