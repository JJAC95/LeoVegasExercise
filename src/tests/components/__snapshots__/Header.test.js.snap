// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders correctly 1`] = `
<div>
  <div
    class="headerItems"
  >
    <div
      class="searchSection"
    >
      <div
        class="searchBarContainer"
      >
        <div
          style="position: relative;"
        >
          <input
            id="input"
            name="searchBar"
            placeholder="Search Games"
            type="text"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
